src/twisted/internet/test/test_unix.py

```diff
@@ -429,18 +443,94 @@ def _dataReceived(self, data):
 
         dataToSend = b'some data needs to be sent'
         fdsToSend = [sendSocket.fileno(), recvSocket.fileno()]
-        ancillary = [(SOL_SOCKET, SCM_RIGHTS, pack('ii', *fdsToSend))]
-        sendmsg(sendSocket, dataToSend, ancillary)
+        ancillary, expectedCount = ancillaryPacker(fdsToSend)
+        try:
+            sendmsg(sendSocket, dataToSend, ancillary)
+        except (socket.error, OSError) as e:
+            # Some platforms fail if ancillary contains more than
+            # one entry (Mac OS 10.9 and 10.10, for example).
+            # No point in continuing if sendmsg fails, skip the test.
+            raise SkipTest('sendmsg failed: %s' % (e,))
```

Comment:
This is an instance of http://xunitpatterns.com/Conditional%20Test%20Logic.html.

Skips should be computed outside of the tests.  The failure mode of computing a skip in-line like this is that a _buggy implementation_ looks the same to us as an _unsupported platform_.  Explicitly check if it's a known-to-not-support-multiple-sendmsg()s platform, then skip the test by setting its `skip` attribute on the outside.